yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -760,7 +756,13 @@ object Client extends Logging {
 
   /**
    * Populate the classpath entry in the given environment map.
-   * This includes the user jar, Spark jar, and any extra application jars.
+   *
+   * User jars are generally not added to the JVM's classpath; those are handled by the AM and
```

Comment:
There's very little code needed to keep this option working; I'd rather leave it around as deprecated for one more release, since even though it's not documented, it's been recommended to people to fix certain issues before.